### ISSUE TITLE
Ctrl-Enter does not send the mail

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/mail#2425]
Usually Ctrl-Enter submits forms or sends messages or sends mails. It’s a quick shortcut to not need to click the "Send" button.

Currently it just inserts a line break, just like Enter.

**Comment by StCyr** [comment]
how should it work on macOS? CTRL-Enter or Meta-Enter?

**Comment by StCyr** [comment]
small study shows that it's not that easy to intercept keypress events with CKEditor.

when using ckeditor's @input event, the event contains the actual ckeditor's content and not the information about the key pressed

Here's what I did:

```javascript
                onTextEditorInputChanged(e) {
                        console.log("onTextEditorInputChanged", e)
                        if (e.keycode == 13 && (navigator.platform.match('Mac') ? e.metaKey : e.ctrlKey)) {
                                this.onSend()
                                return
                        }
                        this.saveDraftDebounced(this.getMessageData())
                },
```

And the result is:

```
1:41:22.965 onTextEditorInputChanged <p>test</p> undefined Composer.vue:378
11:41:29.571 onTextEditorInputChanged <p>test test</p><p>&nbsp;</p> undefined Composer.vue:378
11:41:32.530 onTextEditorInputChanged <p>test test</p><p>test</p> undefined Composer.vue:378
11:41:33.726 onTextEditorInputChanged <p>test test</p><p>testte</p> undefined Composer.vue:378
11:41:34.159 onTextEditorInputChanged <p>test test</p><p>testtest</p> undefined Composer.vue:378
11:41:36.274 onTextEditorInputChanged <p>test test</p><p>testtest</p><p>&nbsp;</p> undefined
```
see also https://ckeditor.com/old/forums/FCKeditor-2.x/changing-behavior-for-Enter-key

**Comment by ChristophWurst** [comment]
Known regression of #1988 https://github.com/nextcloud/mail/blob/4e777e6fbcb3be22ca8ba893fa70f80ff055cc37/src/components/Composer.vue#L104
https://github.com/nextcloud/mail/blob/4e777e6fbcb3be22ca8ba893fa70f80ff055cc37/src/components/Composer.vue#L384-L389

Thanks for having a look @StCyr!

**Comment by jancborchardt** [comment]
> how should it work on macOS? CTRL-Enter or Meta-Enter?

Should be Cmd-Enter (I assume that’s what you mean by Meta-Enter?)

**Comment by StCyr** [comment]
> > how should it work on macOS? CTRL-Enter or Meta-Enter?
> 
> Should be Cmd-Enter (I assume that’s what you mean by Meta-Enter?)

Yes, that's what I meant sorry :-)

So, is the fix as simple as uncommenting the "@keypress="onBodyKeyPress" line (and adding support of Cmd key)?

**Comment by ChristophWurst** [comment]
No, unfortunately the `keypress` even only exists on DOM inputs, not on the CKEditor component.
